ibis/expr/types/relations.py

```diff
@@ -880,25 +880,25 @@ def _ipython_key_completions_(self) -> list[str]:
         return self.columns
 
     @property
-    def columns(self) -> list[str]:
-        """The list of column names in this table.
+    def columns(self) -> tuple[str, ...]:
```

Comment:
I guess this is _technically_ a breaking change, though I'd be surprised if this meaningfully affected downstream usage. :shrug: if wrapping it in a `list` (should be negligible in perf cost for realistic ibis usage) would be better to avoid all the ripple-effect changes. Would certainly make this a shorter PR.

Reply:
My mistake - I thought the cost here was coming from abstractions in going through `self.schema().names`, not the list conversion itself. Nevermind.